src/cargo/core/compiler/job_queue.rs

```diff
@@ -764,6 +797,71 @@ impl<'cfg> DrainState<'cfg> {
                 // It doesn't really matter if this fails.
                 drop(cx.bcx.config.shell().status("Finished", message));
             }
+
+            if !self.future_incompat_reports.is_empty() {
+                self.future_incompat_reports.sort_by_key(|r| r.krate);
+
+                let crates_and_versions = self
+                    .future_incompat_reports
+                    .iter()
+                    .map(|r| format!("`{}` v{}", r.krate, r.version))
+                    .collect::<Vec<_>>()
+                    .join(", ");
+
+                drop(cx.bcx.config.shell().warn(&format!("the following crates contain code that will be rejected by a future version of Rust: {}",
```

Comment:
I'm not sure how to get this into `full_report` while retaining the warning color.